### PR TITLE
Added WIT bindings for OAM types and subscription method

### PR DESCRIPTION
**Pull request by LUK3ARK**
Here I add all bindings for wit generated OAM types and binds them to be translated to and from with the rust types already defined.

I also add a subscribe method which allows the subscription to wadm status api through the client.

NOTE: previously this subscription relied on wrpc and provider specific implementations. I chose to implement as mpsc to keep the wadm client as a thin agnostic wrapper where we can use in multiple streaming scenarios whether that is in CLI, providers, etc. 

Let me know if anyone has any improvements or ideas

### CODE REVIEW

**Comment by LUK3ARK** [comment]
Where is best to keep the WIT for this? Should we just add release to this repo or the main wasmcloud repo? 
The deps.toml file of the wadm provider in the wasmcloud repo will need to be adjusted to reflect this.

**Comment by thomastaylor312** [comment]
@LUK3ARK I think the wit would belong here for now I think for discoverability. We'd need to add a build for packaging up the wit like the one found [here](https://github.com/wasmCloud/wasmCloud/blob/main/.github/workflows/wash-plugins.yml). In the next few months we'll covert it to be stored in OCI, but for now, this would allow it to integrate in with the current flow

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
This is really great! And thanks for the bonus status streaming stuff. Added a few comments for addressing

File: wit/wadm/client.wit [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    get-model-history: func(model-name: string, lattice: option<string>) -> result<list<version-info>, string>;
+
+    // Retrieves the status of a given model by name.
+    get-model-status: func(model-name: string, lattice: option<string>) -> result<status, string>;
+
+    // Retrieves details on a given model.
+    get-model-details: func(model-name: string, version: option<string>, lattice: option<string>) -> result<oam-manifest, string>;
+
+    // Deletes a model version from the WADM system.
+    delete-model-version: func(model-name: string, version: option<string>, delete-all: bool, lattice: option<string>) -> result<bool, string>;
```
Comment (thomastaylor312) [inline wit/wadm/client.wit:36]: Oh heads up that the signature for this now should drop `delete-all` as that is now the default behavior. Everything else can stay the same

File: wit/oam/oam.wit [hunk 2]
```diff
@@ -0,0 +1,93 @@
+package wasmcloud:oam@0.1.0;
```
[the remark this answers is not in the document]
Comment (LUK3ARK) [reply wit/oam/oam.wit]: i just put everything in wadm this good?

File: wit/wadm/client.wit [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package wasmcloud:wadm@0.1.0;
+
+interface client {
+    use wasmcloud:oam/types@0.1.0.{oam-manifest};
```
Comment (thomastaylor312) [inline wit/wadm/client.wit:4]: Looks like you did end up combining the oam stuff into wadm types, so should this be removed from the wit files?

File: wit/oam/oam.wit (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomastaylor312) [reply wit/oam/oam.wit]: Yep, you're good!

**Review by thomastaylor312 — APPROVED** [review]

**Comment by thomastaylor312** [comment]
@LUK3ARK good to go pending a quick rebase!

**Review by thomastaylor312 — APPROVED** [review]

**Review by brooksmtownsend — APPROVED** [review]
🚀 @LUK3ARK looks like this branch just might have some conflicts to resolve and otherwise g2g

**Comment by brooksmtownsend** [comment]
Rebasing...

**Pull request merged**